core/src/main/scala/kafka/server/AddPartitionsToTxnManager.scala

```diff
@@ -85,7 +105,7 @@ class AddPartitionsToTxnManager(
   private val verificationFailureRate = metricsGroup.newMeter(VerificationFailureRateMetricName, "failures", TimeUnit.SECONDS)
   private val verificationTimeMs = metricsGroup.newHistogram(VerificationTimeMsMetricName)
 
-  def verifyTransaction(
+  def addOrVerifyTransaction(
```

Comment:
Previously we were hitting errors on the boundaries of transactions because there weren't the same checks on the client side. Are all of those resolved with the existing epoch checks? Do we return retriable vs. fatal errors at the right times?

Reply:
Are you referring to the UT failures for the previous closed PR?
We still have the problem with testBumpTransactionalEpochAfterInvalidProducerIdMapping after the epoch bump change. I included a fix in the TransactionManager which reset the epochBumpRequired if received a success InitProducerId response.

Reply:
I think this is answered by follow up comments so I will close this to reduce clutter in the PR 😅